src/ui/main_window.py

```
@@ -195,9 +241,24 @@ def _close_experiment(self) -> None:
             return
 
         # Save current ROI to experiment before closing
-        current_roi = self.viewer.get_current_roi()
-        if current_roi is not None:
-            x, y, width, height = current_roi
+        current_roi = None
+        try:
+            current_roi = self.viewer.get_current_roi()
+        except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)